extensions/wikia/UserLogin/js/UserSignupAjaxValidation.js

```diff
@@ -44,14 +38,41 @@
 		);
 	};
 
+	/**
+	 * Call back end for validation of input where user action occurred
+	 * @todo This was copied from validateInput with some added input name mapping. Clean this up in a separate ticket.
+	 * @param {Event} e Browser event like input blur
+	 */
+	UserSignupAjaxValidation.prototype.validateMappedInput = function (e) {
+		var el = $(e.target),
+			paramName = el.attr('name'),
+			params = this.getDefaultParamsForAjax(),
+			mappedParamName;
+
+		if (paramName === 'username') {
```

Comment:
Probably it would be better to create an object with mapped params names and then just use something like jQuery.extend. It may not be shorter than these ifs, but for longer terms it would be much easier to maintain.